webrtc.html

```diff
@@ -7283,25 +7283,24 @@ <h2>Methods</h2>
       <p>The <dfn>RTCRtpContributingSource</dfn> and
       <dfn>RTCRtpSynchronizationSource</dfn> dictionaries contain information
       about a given contributing source (CSRC) or synchronization source (SSRC)
-      respectively, including the most recent time a
-      packet that the source contributed to was played out. The browser MUST
-      keep information from RTP packets received in the previous 10 seconds.
-      When the first audio or video frame contained in an RTP packet is
-      delivered to the <code><a>RTCRtpReceiver</a></code>'s
-      <code><a>MediaStreamTrack</a></code> for playout, the user agent MUST
-      queue a task to update the relevant information for the
-      <code><a>RTCRtpContributingSource</a></code> and
-      <code><a>RTCRtpSynchronizationSource</a></code> dictionaries based on the
-      contents of the packet. The information relevant to the
-      <code><a>RTCRtpSynchronizationSource</a></code> dictionary corresponding
-      to the SSRC identifier, is updated each time, and if the RTP packet
-      contains CSRC identifiers, then the information relevant to the
-      <code><a>RTCRtpContributingSource</a></code> dictionaries corresponding to
-      those CSRC identifiers is also updated.</p>
+      respectively. When an audio or video frame from an RTP packet is delivered
```

Comment:
Note - this text (and the previous) assumes that a frame fits within a single RTP packet. This is true part of the time, and false part of the time (I-frames in particular). It's good if you can change the language to encompass both situations.

Reply:
It seems tricky to define it precisely. There are a couple of cases here:

1. A single frame built by multiple RTP packets with different RTP timestamps. This may happen for audio. It makes sense to process them in ascending RTP timestamp order so that the last packet "wins" for fields like `audioLevel`.

2. A single frame built by multiple RTP packets with the exact same RTP timestamps. This may, as you say, happen for video. Certain values may only be found in a RTP header extension that is only sent with either the first or last packet in a frame. So an implementation would likely want to process the RTP packets as a "group of packets" without any particular order.

3. Hybrid case where both (1) and (2) happens for a single frame that is delivered to `MediaStreamTrack`. Doesn't happen in practice today.

I've updated this PR with "one or more RTP packets" and "MUST process RTP packets in order of ascending RTP timestamps".

Reply:
RTP timestamps are the timestamp of the frame, so if you have "a single (video) frame built by multiple RTP packets with different RTP timestamps", you have a bogus encoder.
The term "frame" is sometimes used with audio to describe a block of samples (20 ms or so), in which case you could imagine those coming from packets with different RTP timestamps.